launch_ros/launch_ros/actions/node.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the Node action."""
+
+import logging
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Text
+
+from launch.action import Action
+from launch.actions import ExecuteProcess
+from launch.launch_context import LaunchContext
+from launch.some_substitutions_type import SomeSubstitutionsType
+from launch.substitutions import LocalSubstitution
+from launch.utilities import normalize_to_list_of_substitutions
+from launch.utilities import perform_substitutions
+
+from launch_ros.substitutions import ExecutableInPackage
+
+from rclpy.validate_namespace import validate_namespace
+from rclpy.validate_node_name import validate_node_name
+
+_logger = logging.getLogger(name='launch_ros')
+
+
+class Node(ExecuteProcess):
```

Comment:
The reason I departed from `Execute` in these is because of how I planed to reuse them as arguments to the "node container", basically something like this:

```
LaunchDescription([
  ComponentNodeContainer(name='whatever', nodes=[
     Node(...),
     Node(...), 
  ]),
  Node(...),
])
```

Where the first two `Node` would never be visited, and instead would be introspected by the `ComponentNodeContainer` to figure out how to load them dynamically, and the last `Node` would actually get visited and therefore executed.

Reply:
Maybe there's a better way to organize the names, with that in mind. I don't know.

I also thought about an `Execute` action which would look like:

```python
LaunchDescription([
  ComponentNodeContainer(name='whatever', nodes=[
     Node(...),
     Node(...), 
  ]),
  Execute(executable=Node(...), cwd='something'),
])
```

In which case `Node` would no longer inherit from `ExecuteProcess`.

Reply:
Even if `Node` doesn't inherit from ExecuteProcess we'd still probably benefit from distinguishing it from something that is actually a node (it is not, in any of these cases). For now, it's namespaced enough that I suppose we can lean on that.

Reply:
That's part of the reason I use `launch.x.Y` in most of the demos, to reinforce those differences.

I also somewhat inconsistently use `Launch<thing>` and `<thing>`, e.g. `LaunchDescription` versus `Action` (and not `LaunchAction`). I don't have a good reason for the inconsistency, other than I felt that in context `launch.Action` was specific enough, but `launch.Description` was not (mostly due to where and how it's used in the examples.

It might be something we want to revisit in the near future.

Do you want me to change anything for this pr related to this conversation?

Reply:
I think for this PR we're good, and then there's a chance we'll rename things in the future.